homeassistant/components/group/sensor.py

```diff
@@ -328,11 +332,21 @@ def async_update_group_state(self) -> None:
         states: list[StateType] = []
         valid_states: list[bool] = []
         sensor_values: list[tuple[str, float, State]] = []
+        (
+            self.entities_can_convert,
+            converter,
+            base_uom,
+        ) = self._check_entities_device_class_and_uom()
```

Comment:
Maybe this can be called a single time when first creating the sensor, later on we just call the converter and catch errors if the units are no longer matching?
Also, I think the config flow could block grouping sensors with conflicting device classes / units.
Note that the sensor base class will soon enforce the unit to be valid:
https://github.com/home-assistant/core/blob/a9f7b3236d53de85beec00b540a44a4877d41776/homeassistant/components/sensor/__init__.py#L658-L674

Reply:
So should we allow for combining entities with and w/o device_class but not mixing between different device_classes?
Should we then also remove the device_class and UoM from config entry options so it's always taken from the input entities?

Reply:
So here's my proposal also considering some previous comments

Main question is perhaps if 2 below should even be allowed but probably will break someone's setup --> breaking change.
Matching @emontnemery comment above and also "aligns" with what is changing with `sensor` anyhow.

Check devices class of input sensors during setup.
1. If same device_class and/or without device_class go to calculation and check if matching UoM's. Set base device_class and UoM from first hit.
2. If not same device_class, don't set device_class and UoM or use user configured (if using YAML setup).

During runtime:
1. If 1 above, do conversions. Fail conversions and state if UoM change outside of allowed according to device_class
2. If 2 above, no conversions. Just set value.